aspnetcore/grpc/versioning.md

```diff
@@ -0,0 +1,93 @@
+
+* Is not a protocol breaking change.
+* Setting an error condition if the new field is not set makes it a breaking change for old clients.
+
+### Non-breaking changes
+
+These changes are non-breaking at a gRPC protocol level, and .NET binary level.
+
+- **Adding a new service**
+- **Adding a new method to a service**
+- **Adding a field to a request message** - Fields added to a request message are deserialized with the [default value](https://developers.google.com/protocol-buffers/docs/proto3#default) on the server when not set.
+- **Adding a field to a response message** - Fields added to a response message are deserialized into the message's [unknown fields](https://developers.google.com/protocol-buffers/docs/proto3#unknowns) collection on the client.
+- **Adding a value to an enum** - Enums are serialized as a numeric value. New enum values are deserialized on the client to the enum value without an enum name.
+
+### Binary breaking changes
+
+The following changes are non-breaking at a gRPC protocol level, but the client needs to be updated if it upgrades to the latest *.proto* contract or client .NET assembly.
+
+- **Removing a field** - Values from a removed field are deserialized to a message's [unknown fields](https://developers.google.com/protocol-buffers/docs/proto3#unknowns). This isn't a gRPC protocol breaking change, but the client needs to be updated if it upgrades to the latest contract. It is important that a removed field number isn't accidentally reused in the future. One way to make sure this doesn't happen is to specify deleted field numbers and names on the message using Protobuf's [`reserved`](https://developers.google.com/protocol-buffers/docs/proto3#reserved) keyword.
+- **Renaming a field** - Field names are only used in generated code. The field number is used to identify fields on the network. The client will need to be updated if it upgrades to the latest contract.
+- **Renaming a message** - Message names are not sent on the network so this isn't a gRPC protocol breaking change, but the client will need to be updated if it upgrades to the latest contract.
+
+### Breaking changes
+
+These are protocol and binary breaking changes.
+
+- **Changing a field data type** - Changing a field's data type to an [incompatible type](https://developers.google.com/protocol-buffers/docs/proto3#updating) will cause errors when deserializing the message. Even if the new data type is compatible, it is likely the client will need to be updated to support the new type if it upgrades to the latest contract.
+- **Changing a field number** - The field number is used to identify fields on the network.
+- **Renaming a package, service or method** - gRPC uses the package name, service name and method name to build the URL. The client gets an *UNIMPLEMENTED* status from the server.
+- **Removing a service or method** - The client gets an *UNIMPLEMENTED* status from the server when calling the removed method.
+
```

Comment:
What would you think about adding some additional stuff here about breaking changes that aren't protocol or codegen changes.
- Expanding the allowed range of a return-value (adding an enum value)
- Adding additional validate rules

Reply:
I think that is a never ending subject. At the top of this section I say that you need to determine whether older clients can logically still run.